bluesky/run_engine.py

```diff
@@ -807,12 +810,18 @@ def _open_run(self, msg):
 
     @asyncio.coroutine
     def _close_run(self, msg):
+        if not self._run_is_open:
+            raise IllegalMessageSequence("A 'close_run' message was received "
+                                         "but there is no run open. If this "
+                                         "occurred after a pause/resume, add "
+                                         "a 'checkpoint' message after the "
+                                         "'close_run' message.")
```

Comment:
If close_run always creates a checkpoint, then we rule out the possibility of placing checkpoints, say, between every five runs of a multi-sample plan. A more useful thing to do would be to insert a checkpoint at the end of a run *if and only if that run contained a check point*. But I guess I would rather do this with a "preprocessor" than in the RunEngine. So far the RunEngine is not in the business of fixing plans on the fly, and doing so would add even more state.